### ISSUE TITLE
Feature Request: Add a PK field to prisma generated mapping tables

### ISSUE THREAD

**Issue by webnoob** [prisma/prisma#11110]
Hi,

I'm using a managed DB hosting provider (OVH) and in turn have no control over the `sql_require_primary_key` setting. When importing the prisma generated DB, I get an error saying each table must have a primary key. For the most part, this is fine but I'm struggling with the tables that prisma creates as mapping tables.

Take this model:

```
model User {
  id          Int      @id @default(autoincrement())
  canActAs    User[] @relation("actingAs", references: [id])
  hasActingAs User[] @relation("actingAs", references: [id])
}
```

Prisma will create a `_actingAs` table with 2 columns, `A` and `B`. I need to be able to add a PK field to that table.

Note: I tried creating an empty migration which altered those mapping tables adding a PK field but it created a new migration removing it after because the schema was out of line.

Note: Initially, I didn't think those tables were causing my error because when I  look at the `_actingAs` mapping table, I do see that both columns `A` and `B` are down as PK fields:
![image](https://user-images.githubusercontent.com/10404747/148643815-333df5c0-3f88-4789-b926-7ec00cb567c5.png)
 but the following query raises them as not having a PK field:

```
select tab.table_schema as database_name,
       tab.table_name
from information_schema.tables tab
left join information_schema.table_constraints tco
          on tab.table_schema = tco.table_schema
          and tab.table_name = tco.table_name
          and tco.constraint_type = 'PRIMARY KEY'
where tco.constraint_type is null
      and tab.table_schema not in('mysql', 'information_schema', 
                                  'performance_schema', 'sys')
      and tab.table_type = 'BASE TABLE'
--      and tab.table_schema = 'sakila' -- put schema name here
order by tab.table_schema,
         tab.table_name;
```

_Originally posted by @webnoob in https://github.com/prisma/prisma/discussions/11028_

**Comment by janpio** [comment]
This bit is confusing me a bit:
> Note: Initially, I didn't think those tables were causing my error because when I look at the _actingAs mapping table, I do see that both columns A and B are down as PK fields:

Can you share the SQL of the affected table here so it is easier for us to reproduce than to run `db push` or Migrate?

**Comment by janpio** [comment]
Yes, it seems that MySQL workbench accepts the "UNIQUE KEY" of that table to define a "PK", but in reality (and for OVH or `sql_require_primary_key`) that is not really the case - hence causing trouble.

**Comment by webnoob** [comment]
Indeed, that's my assessment too. It was baffling me at first.

**Comment by loren** [comment]
I'm running into the same issue but on Postgres14. Manually creating the composite primary key satisfies my other app that requires a PK for introspection:
```
alter table "_FooToBar" add PRIMARY KEY("A","B");
```
But then Prisma complains and wants to reset the DB. 

Reading through the docs, I do not see a workaround for this (like `@@ignore`), but please let me know if there is. I understand I could use an explicit relation as a workaround, but I am trying to avoid this route if I can.

I would like to suggest having this primary key created by Prisma as part of the DDL that creates the implicit many-to-many table. I'd be happy to take a crack at a PR if someone can point me to the general spot in the codebase.

**Comment by ekansh005** [comment]
Any update on this one? 
PgAdmin as well as Supabase table editor, both don't allow me to delete the relation I just created (with Supabase table editor).

**Comment by samuelcole** [comment]
i believe this was resolved by https://github.com/prisma/prisma-engines/pull/5057
